xcube_stac/store.py

```diff
@@ -82,391 +93,162 @@ def __init__(
             self._searchable = False
         self._catalog = catalog
 
-        # TODO: Add a data store "file" here when implementing
-        # open_data(), which will be used to open the hrefs
+        self._storage_options_s3 = storage_options_s3
+
+        self._stack_mode = stack_mode
+        if stack_mode is False:
+            self._impl = SingleImplementation(
+                self._catalog, self._url_mod, self._searchable, self._storage_options_s3
+            )
+        elif stack_mode is True or stack_mode == "odc-stac":
+            self._impl = StackImplementation(
+                self._catalog, self._url_mod, self._searchable, self._storage_options_s3
+            )
```

Comment:
Else? Since it is possible to instantiate the store via ctor, you should raise a `ValueError()` in the else-case. Otherwise `_impl` will not be initialised.

Reply:
The data store parameters are  validated against the schema in the xcube data store framework. the schema is define in [constants.py](https://github.com/xcube-dev/xcube-stac/blob/konstntokas-xxx-implement_stac_mode/xcube_stac/constants.py#L74) and this is tested in (test_store.py)[https://github.com/xcube-dev/xcube-stac/blob/konstntokas-xxx-implement_stac_mode/test/test_store.py#L90].

 However, I agree, if a user would initialize a stac data store directly by calling the class `StacDataStore`, then this is not covered. I therefore added a else case.